connector_magento_configurable/models/product_attribute_line/importer.py

```diff
@@ -0,0 +1,133 @@
+        line = self._get_magento_product_attribute_line(
+            attribute,
+            value,
+            record
+        )
+        self._import_record(line, job_options={'priority': 100})
+
+    def run(self, filters=None):
+        """ Run the synchronization """
+        record = filters['record']
+        updated_variants = self.backend_adapter.list_variants(
+            record.default_code)
+        available_attributes = self.env[
+            'magento.product.attribute'].search_read([], [
+                'name',
+                'odoo_id',
+            ])
+        value_binder = self.binder_for('magento.product.attribute.value')
+        product_binder = self.binder_for('magento.product.product')
+        for variant in updated_variants:
+            magento_product = product_binder.to_internal(
+                variant['entity_id'], unwrap=False)
+            if not magento_product:
+                raise MappingError("The product with "
+                                   "magento id %s is not imported." %
+                                   variant['entity_id'])
+            attribute_value_ids = []
+            for attribute in available_attributes:
+                if variant.get(attribute['name']):
+                    value = value_binder.to_internal(
+                        variant[attribute['name']], unwrap=False)
+                    if not value:
+                        raise MappingError("The product attribute value with "
+                                           "magento id %s is not imported." %
+                                           variant[attribute['name']])
+                    self._import_magento_product_attribute_line(
+                        record,
+                        variant,
+                        attribute,
+                        value,
+                    )
+                    attribute_value_ids.append((4, value.odoo_id.id))
+            if attribute_value_ids:
+                self._write_product(
+                    magento_product,
+                    record.product_tmpl_id.id,
+                    attribute_value_ids,
+                    )
+
+
+class ProductAttributeLineImporter(Component):
+    _name = 'magento.product.attribute.line.importer'
+    _inherit = 'magento.importer'
+    _apply_on = ['magento.product.attribute.line']
+
+    def _get_magento_data(self, storeview_id=None):
+        """
+        In this case,
+        the magento_record already contains all the data to insert,
+        no need to make a xmlrpc call
+        """
+        return self.magento_record
+
+    def run(self, magento_record, force=False):
+        self.magento_record = magento_record
+        super(ProductAttributeLineImporter, self).run(
+            magento_record['external_id'],
+            force,
+            )
+        self.env['product.template'].search([
+            ('product_variant_ids', '=', False)
+        ]).with_delay(priority=101, eta=60).delayable_unlink()
```

Comment:
Is there a reason for giving an eta in 60 seconds?

Reply:
The goal is to avoid trying to unlink a template still having product.product linked to it. Second thought, not sure if this is necessary but didn't think of another way to avoid the problem.